### DISCUSSION TITLE
Lost all devices after removing them from tuya smart life app.

### DISCUSSION THREAD

**Discussion by mrsofa** [make-all/tuya-local#2318] [General]
I blocked the internet connection and assigned a static local IP for all of my Tuya wifi devices. I also deleted the devices from the Tuya Smart life app. And now my devices are unreachable.
Is there a way to get them back? Enabling back the internet and adding them again in Smart life app is possible but then I have to add them again in the Tuya local integration. Which results in loosing all of my history stats and I have to make everything from scratch in the front end.

I didn't want to make this post as a bug because maybe I missed a part where it says I must not delete my devices in the tuya local app. If that is the case it looks as it this integration isn't 100% local and if for some reason Tuya closes it's cloud or deletes the app all of my Tuya devices will be unusable.

Thank you

**Comment by make-all** [comment]
If the devices know (somehow, maybe they got a brief connection to the server at some point, or communicated directly with the app when they were removed) they have been deleted from the app, they will reset into pairing mode and your local keys will no longer be valid.

Some devices will also refuse local connections if the internet is unavailable but they can get the server address through DNS - for those devices, blocking their access to DNS might help.

As long as the device ids remain the same (it seems some newer devices also reset these when re-pairing, but in general for older devices the device ids are fixed), you can just update the local key in the tuya-local config once they have been re-paired with the app.

**Reply by mrsofa** [reply]
It's possible to block access to dns for a specific device? Is that something i can do in the port sharing section on my modem?

I disabled the old device, went to add the new one via tuya cloud asissted setup, copied the local key, canceled the setup, enabled the old device, added the new local key, the ip is the same and it still fails: Unable to connect to your device with those details. It could be an intermittent issue, or they may be incorrect.

**Reply by make-all** [reply]
I think it is quite advanced, and maybe involves setting up your own internal DNS server that is configured to send different replies to different devices. Or you can maybe set up a firewall rule to block port 53 UDP and TCP for that device.
It's not something I do myself.

**Reply by mrsofa** [reply]
Okay I will block port 53 then, that's what I needed thx.

About adding the devices, again. Can i somehow edit the current ones manually in some sort of .yaml file? I want to try to manually change the local keys and restart everything.

**Reply by make-all** [reply]
You can use the UI to edit the settings.  There is no bulk change tool, editing the HA config files directly could introduce inconsistencies, as the information is duplicated and spread through several files.
